examples/e-commerce/src/App.vue

```diff
@@ -517,6 +519,10 @@ export default {
         value.max !== null ? value.max : range.max,
       ];
     },
+    getSelectedHitsPerPageValue() {
+      const [, hitsPerPage] = document.location.search.match(/hitsPerPage=([0-9]+)/) || [];
```

Comment:
are we sure we want to solve this like this? It seems quite messy to me to rely on the URL here

Reply:
Why is it messy? Do you have a better solution in mind?

Reply:
Widgets themselves shouldn't be taking the URL into account, since that's handled by routing. How is this solved in the other flavors?

Reply:
The issue is that the `HitsPerPage` widget [only uses the `default` property on `items` to find the selected one](https://github.com/algolia/vue-instantsearch/blob/ede58524399dce43f6107d4e8d6be98417f29bbd/src/components/HitsPerPage.vue#L57). It does not take the current state into account ( like [`react-instantsearch` for example](https://github.com/algolia/react-instantsearch/blob/4906170382232d9c4eed3a5723f544e63eff9e0a/packages/react-instantsearch-core/src/connectors/connectHitsPerPage.js#L13-L27) ). Which forces us to retrieve this value by ourselves.

Reply:
I think in this case the test should be failing instead of passing with this workaround. The behaviour is actually wrong.

Reply:
we can skip the test for vue here?

Reply:
I'd rather keep this small workaround for the demo than removing a test for Vue then working on a fix for `HitsPerPage` (should be something quick to fix).

Reply:
if that's what you prefer